src/nodes.coffee

```diff
@@ -281,9 +281,15 @@ exports.Base = class Base
   # **WARNING: DO NOT OVERRIDE THIS METHOD IN CHILD CLASSES.**
   # Only override the component `ast*` methods as needed.
   ast: (o, level) ->
+    # Merge `level` into `o` and perform other universal checks.
     o = @astInitialize o, level
+    # Create serializable representation of this node.
     astNode = @astNode o
-    @astAddReturns astNode
+    # Mark AST nodes that correspond to expressions that (implicitly) return.
+    # We can’t do this as part of `astNode` because we need to assemble child
+    # nodes first before marking the parent being returned.
+    astNode.returns = yes if @astNode? and @canBeReturned
```

Comment:
If you want this to be more obviously a modification of a JSON-style object/"`Object.assign()`-y", you could write like eg:
```
if @astNode? and @canBeReturned
  Object.assign astNode, {returns: yes}
```